src/app/search/search-entry-table.ts

```diff
@@ -33,6 +33,32 @@ export abstract class SearchEntryTable extends Base implements OnInit {
 
   public readonly displayedColumns = ['name', 'verified', 'author', 'descriptorType', 'projectLinks', 'starredUsers'];
   abstract dataSource: MatTableDataSource<Workflow | DockstoreTool>;
+
+  static compareAttributes(a: DockstoreTool | Workflow, b: DockstoreTool | Workflow, attribute: string, direction: string) {
+    let aVal = a[attribute];
+    let bVal = b[attribute];
+    console.log(attribute);
+
+    // regardless of sort direction, null or empty values go at the end
+    if (!aVal) {
+      return 1;
+    }
+    if (!bVal) {
+      return -1;
+    }
+
+    // ignore case when sorting by author or name
+    if (attribute === 'author' || attribute === 'name') {
+      aVal = aVal.toLowerCase();
+      bVal = bVal.toLowerCase();
+    }
+    if (direction === 'asc') {
+      return aVal < bVal ? -1 : 1;
```

Comment:
I figured the results would be the same because even if they're equal they still have to get put in some order, so it wouldn't matter? But yes returning 0 would be more technically correct

Reply:
In looking around I've seen it both ways. so it might be OK. But looking at [the Mozilla doc on sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort), their examples return 0 (while having a note about certain browsers not honoring it equally). I think it best to return 0.

Depending on the implementation of the sort, and this is just a wild guess on my part, I would think 0 could be marginally faster, albeit probably not noticeably so. If you return 0, you know you don't have to sort the item any more, if you return non-zero, then you might need to sort the item against another item.